### PR TITLE
Bintray Deprecated. Updated to new official repo. 

### PR DESCRIPTION
**Pull request by MoonshineJack**
# Description

Due to Bintray deprecation, update cookbook to leverage the new official repo(S) for Rundeck. In the process, discovered various failures in testing, which have been addressed.  

## Issues Resolved

N/A

## Check List

- [X] All tests pass. See TESTING.md for details.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Review by Xorima — DISMISSED** [review]

File: resources/server_install.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@
 property :exec_logdir, String, default: lazy { "#{basedir}/logs" }
 property :extra_wait, Integer, default: 120
 property :framework_properties, Hash, default: {}
-property :grails_port, Integer, default: lazy { use_ssl ? 443 : 80 }
+property :grails_port, Integer, default: lazy { use_ssl ? 443 : 4440 }
```
Comment (Xorima) [inline resources/server_install.rb:33]: Not a rundeck user but why port 4440 instead of 80?

Comment (codayblue) [reply resources/server_install.rb:33]: That is the default port. The port 80 was causing the ruby block to error. I assume it worked at some point in time but now it was just causing errors.

File: resources/repository.rb [hunk 2]
```diff
@@ -20,19 +20,16 @@
...
-property :gpgkey, String, default: lazy {
-  if platform_family?('rhel')
-    'http://rundeck.org/keys/BUILD-GPG-KEY-Rundeck.org.key'
-  else # 'debian'
-    'https://bintray.com/user/downloadSubjectPublicKey?username=bintray'
-  end
-}
-property :gpgcheck, [true, false], default: true
+property :gpgkey, String, default: 'https://packages.rundeck.com/pagerduty/rundeck/gpgkey'
+
```
Comment (ramereth) [inline resources/repository.rb:29]: Can you please remove these extra new lines here and below in the property declarations?

File: resources/repository.rb [hunk 3]
```diff
@@ -49,7 +46,9 @@
 
     apt_repository 'rundeck' do
       uri new_resource.package_uri
-      distribution '/'
+      deb_src new_resource.debsrc
+      components %w(any main)
+      distribution ''
```
Comment (ramereth) [inline resources/repository.rb:51]: Is this needed or can it just be removed?

File: resources/server_install.rb [hunk 4]
```diff
@@ -340,6 +340,12 @@
     notifies :run, 'ruby_block[wait for rundeckd startup]', :immediately
   end
 
+  execute 'enable rundeckd' do
+    command 'systemctl enable rundeckd'
+    action :run
+    only_if { node['init_package'] == 'systemd' }
+  end
```
Comment (ramereth) [inline resources/server_install.rb:347]: This should use the `service` resource instead of `execute`.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: resources/server_install.rb (hunk 4 above)
Comment (codayblue) [reply resources/server_install.rb:347]: @ramereth when we tried service is was reporting that the service was enabled but it was still disabled in systemd. This was the work around to fix it.

File: resources/repository.rb (hunk 3 above)
Comment (MoonshineJack) [reply resources/repository.rb:51]: This is necessary.

File: resources/server_install.rb (hunk 4 above)
Comment (Xorima) [reply resources/server_install.rb:347]: @lamont-granquist / @tas50  recommendations for this?

Comment (MoonshineJack) [reply resources/server_install.rb:347]: Building on what @codayblue said, this only occurs on Debian distros.

**Comment by ramereth** [comment]
@MoonshineJack can you please allow maintainers access to push for this PR? I have some local fixes which will address the issues I'd like to push up.

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [6.0.0](https://github.com/sous-chefs/rundeck/releases/tag/6.0.0)

**Pull request merged**